app/services/moderator/manage_activity_and_roles.rb

```diff
@@ -70,6 +70,18 @@ def handle_user_status(role, note)
         remove_negative_roles
         user.remove_role :pro
         add_trusted_role
+      when "Admin"
+        check_super_admin
+        remove_negative_roles
+        user.add_role :admin
+      when "Super Admin"
+        check_super_admin
+        remove_negative_roles
+        user.add_role :super_admin
+      when /^(Resource Admin: )/
+        check_super_admin
+        remove_negative_roles
+        user.add_role(:single_resource_admin, role.split("Resource Admin: ").last.safe_constantize.new.class)
```

Comment:
After `safe_constantize` you already have the class, no need for `.new.class`:

```ruby
"User".safe_constantize == "User".safe_constantize.new.class
#=> true
```

```suggestion
        user.add_role(:single_resource_admin, role.split("Resource Admin: ").last.safe_constantize)
```